src/DynamoCoreWpf/UI/HeaderTemplateSelector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using Dynamo.UI.Controls;
+
+namespace Dynamo.Controls
+{
+    public class HeaderTemplateSelector : DataTemplateSelector
```

Comment:
Some screenshots:
![image](https://cloud.githubusercontent.com/assets/8158551/5615956/9551aa5e-9508-11e4-88fb-eb9c3f363db3.png) ![image](https://cloud.githubusercontent.com/assets/8158551/5615961/a8a501aa-9508-11e4-8986-51dc689da918.png)

So as you can notice, there is a different design [and behavior] for one header and a number of headers. That's why I am using two templates: for one header and for many headers.